src/files-and-uploads/FilesAndUploads.jsx

```diff
@@ -0,0 +1,273 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import _ from 'lodash';
+import { injectIntl, FormattedMessage, intlShape } from '@edx/frontend-platform/i18n';
+import {
+  DataTable,
+  TextFilter,
+  CheckboxFilter,
+  Dropzone,
+  CardView,
+  useToggle,
+} from '@edx/paragon';
+import Placeholder, { ErrorAlert } from '@edx/frontend-lib-content-components';
+
+import { RequestStatus } from '../data/constants';
+import { useModels } from '../generic/model-store';
+import {
+  addAssetFile,
+  deleteAssetFile,
+  fetchAssets,
+  updateAssetLock,
+} from './data/thunks';
+import messages from './messages';
+
+import FileInput, { fileInput } from './FileInput';
+import FilesAndUploadsProvider from './FilesAndUploadsProvider';
+import {
+  GalleryCard,
+  ListCard,
+  TableActions,
+} from './table-components';
+import ApiStatusToast from './ApiStatusToast';
+
+const FilesAndUploads = ({
+  courseId,
+  // injected
+  intl,
+}) => {
+  const dispatch = useDispatch();
+  const defaultVal = 'card';
+  const columnSizes = { xs: 12, sm: 6, lg: 3 };
+  const [currentView, setCurrentView] = useState(defaultVal);
+  const [isDeleteOpen, setDeleteOpen, setDeleteClose] = useToggle(false);
+  const [isAddOpen, setAddOpen, setAddClose] = useToggle(false);
+  const [selectedRowCount, setSelectedRowCount] = useState(0);
+  const assetIds = useSelector(state => state.assets.assetIds);
+
+  useEffect(() => {
+    dispatch(fetchAssets(courseId));
+  }, [courseId]);
+  const totalCount = useSelector(state => state.assets.totalCount);
+  const addAssetStatus = useSelector(state => state.assets.addingStatus);
+  const deleteAssetStatus = useSelector(state => state.assets.deletingStatus);
+  const saveAssetStatus = useSelector(state => state.assets.savingStatus);
+  const loadingStatus = useSelector(state => state.assets.loadingStatus);
```

Comment:
Nit: this will save a lot of time
```suggestion
  const {totalCount , addAssetStatus: addingStatus, deleteAssetStatus: deleteingStatus, ... } = useSelector(state=> state.assets)
```